src/lamplib/src/genny/tasks/auto_tasks_all.py

```diff
@@ -32,36 +36,52 @@ def get_all_builds(global_expansions: dict[str, str], project_file_path: str) ->
             all_builds.append(CurrentBuildInfo(build_info_dict))
     return all_builds
 
-def create_configuration(repo: Repo, builds: List[CurrentBuildInfo], no_activate: bool):
+
+def create_configuration(
+    repo: Repo, builds: List[CurrentBuildInfo], no_activate: bool, activate_tasks: List[VariantTask]
+):
     all_tasks = repo.all_tasks()
     config = Configuration()
     ConfigWriter.configure_all_tasks_modern(config, all_tasks)
-    activate_param = False if no_activate else None
+    activate_all_param = False if no_activate else None
     for build in builds:
 
         build_tasks = repo.variant_tasks(build)
         if len(build_tasks) > 0:
             SLOG.info(f"Generating auto-tasks for variant: {build.variant}")
-            ConfigWriter.configure_variant_tasks(config, build_tasks, build.variant, activate=activate_param)
+            ConfigWriter.configure_variant_tasks(
+                config,
+                build_tasks,
+                build.variant,
+                activate_all=activate_all_param,
+                activate_tasks=activate_tasks,
+            )
         else:
             SLOG.info(f"No auto-tasks for variant: {build.variant}")
     return config
 
+
 def main(project_files: List[str], workspace_root: str, no_activate: bool) -> None:
     reader = YamlReader()
     try:
         global_expansions = reader.load(workspace_root, "expansions.yml")
     except FileNotFoundError:
-        SLOG.error(f"Evergreen expansions file {os.path.join(workspace_root, 'expansions.yml')} does not exist. Ensure this file exists and that it is in the correct location.")
+        SLOG.error(
+            f"Evergreen expansions file {os.path.join(workspace_root, 'expansions.yml')} does not exist. Ensure this file exists and that it is in the correct location."
+        )
         sys.exit(1)
     execution = int(global_expansions["execution"])
+    activate_tasks = set(
+        VariantTask(*item.strip().split(":"))
+        for item in global_expansions.get("activate_generated_tasks", "").split(",")
+    )
```

Comment:
While correct, this is a bit hard to parse. My personal preference would be to split this into 2-3 lines so it reads can be read top down, while right now it needs to be read "bottom-up". Feel free to ignore this.

Reply:
Agreed. Will clean up.